app/models/user.rb

```diff
@@ -51,6 +51,7 @@ class User < ActiveRecord::Base
 
   before_validation :default_display_name, on: [:create, :update]
   before_validation :setup_unsubscribe_token, on: [:create]
+  before_validation :update_ouroboros_created, on: [:create]
```

Comment:
You're correct. The validation won't be called on login. Since they're being skipped anyway doing this as a before_save should work. I think I'm actually going to do it in both places so users created through this method can still be references in other places (like being assigned as collaborators).